File: models/Session.js
const mongoose = require('mongoose');

const playerSchema = new mongoose.Schema({
  queueNumber: { type: Number, required: true },
  name: { type: String, required: true, trim: true, minlength: 2 },
  phone: { type: String, default: '', match: /^\d{10,15}$/ },
  paid: { type: Boolean, default: false },
  addedAt: { type: Date, default: Date.now }
});

const pendingSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true, minlength: 2 },
  phone: { type: String, default: '', match: /^\d{10,15}$/ },
  requestedAt: { type: Date, default: Date.now }
});

const sessionSchema = new mongoose.Schema({
  date: { type: Date, default: Date.now },
  status: { type: String, enum: ['open', 'closed'], default: 'open' },
  shareCode: { type: String, required: true, unique: true },
  queue: [playerSchema],
  pending: [pendingSchema]
});

module.exports = mongoose.model('Session', sessionSchema);

File: app/models/session.py
from datetime import datetime, timezone
from typing import Literal

from beanie import Document, PydanticObjectId
from pydantic import BaseModel, Field
from pymongo import ASCENDING, IndexModel


class Player(BaseModel):
    id: PydanticObjectId = Field(default_factory=PydanticObjectId)
    queue_number: int
    member_id: PydanticObjectId
    paid: bool = False
    added_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class PendingRequest(BaseModel):
    id: PydanticObjectId = Field(default_factory=PydanticObjectId)
    member_id: PydanticObjectId
    requested_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class Session(Document):
    date: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    status: Literal["open", "closed"] = "open"
    share_code: str
    fee: float = 0.0
    title: str = ""
    notes: str = ""
    location: str = ""
    capacity: int | None = None
    queue: list[Player] = []
    pending: list[PendingRequest] = []

    class Settings:
        name = "sessions"
        indexes = [
            IndexModel([("share_code", ASCENDING)], unique=True),
            IndexModel([("status", ASCENDING)]),
            IndexModel([("date", ASCENDING)]),
        ]
